include/cutlass/gemm/collective/xe_mma_mixed_input.hpp

```diff
@@ -459,66 +483,92 @@ struct CollectiveMma<
     static_assert(size_v<LayoutIn> == cosize_v<LayoutIn>);
     static_assert(size_v<LayoutOut> == cosize_v<LayoutOut>);
     static_assert(std::is_same_v<typename EngineOut::value_type, typename EngineScales::value_type>);
-    static_assert(std::is_same_v<LayoutScales, LayoutZeros>);
 
     using SrcType = typename EngineIn::value_type;
     using DstType = typename EngineOut::value_type;
+    using ScaleType = typename EngineScales::value_type;
+    using ZeroType = typename EngineZeros::value_type;
+    if constexpr (KernelConversionMode == ConversionMode::DirectConvert){
```

Comment:
Adding this `if constexpr` means for `ConvertAndScale` and `ConvertAndScaleWithZero` modes, we won't upscale the narrow data type using `NumericArrayConverter` or `convert_FP8_to_FP16`. Instead we rely on element-wise `static_cast`. Is there a reason for this?